common/flatpak-dir.c

```diff
@@ -6135,10 +6140,18 @@ flatpak_dir_install (FlatpakDir          *self,
 
           flatpak_flags |= FLATPAK_PULL_FLAGS_SIDELOAD_EXTRA_DATA;
 
+#ifdef FLATPAK_ENABLE_P2P
+          if (collection_id == NULL &&
```

Comment:
AFAIU, the `collection_id` is effectively only used/cared for when p2p is enabled anyway, so why not simply have this change without the `ifdef`? Would skip repeating a block.

Reply:
In the rest of the codebase the collection ID is ignored if flatpak is compiled without P2P support. Look at `flatpak_dir_pull` for example

Reply:
@mwleeds Do you mean ignored by the preprocessor? In this function alone `collection_id` is used a bunch of times outside of `ifdef`, so IMO we could do the same and skip having to repeat this block (and others) which doesn't get us anything.

Reply:
Looking again I see that `repo_get_remote_collection_id` returns NULL if P2P is disabled, so I pushed a fixup to remove the extra ifdefs. More generally though I don't think enabling collection IDs without enabling P2P is a well tested configuration, so we might want to remove the configure flag before pushing collection IDs on everyone.